pkg/action.go

```diff
@@ -125,31 +136,58 @@ func main() {
-	cds, err := hiveset.HiveV1().ClusterDeployments("").List(context.TODO(), v1.ListOptions{})
-	for _, clusterDeployment := range cds.Items {
-		if (OptIn == "true" && clusterDeployment.Labels["hibernate"] == "true") || (OptIn != "true" && clusterDeployment.Labels["hibernate"] != "skip") {
-			if string(clusterDeployment.Spec.PowerState) != TakeAction {
-
-				fmt.Print(TakeAction + ": " + clusterDeployment.Name)
-				newPowerState := powerStateChange(hiveset, &clusterDeployment, TakeAction)
-
-				// Check the new state and report a response
-				if newPowerState == TakeAction {
-					fmt.Println("  ✓")
-					fireEvent(kubeset, &clusterDeployment, "hibernating", "The cluster "+clusterDeployment.Name+" has powerState "+TakeAction, TakeAction, "Normal")
-				} else {
-					fmt.Println("  X")
-					//"failedhibernating", "The cluster " + clusterName + " did not set powerState to Hibernating", "failedHibernating", "Warning"
-					fireEvent(kubeset, &clusterDeployment, "hibernating", "The cluster "+clusterDeployment.Name+" did not set powerState to "+TakeAction, "failedHibernating", "Warning")
-				}
+	podNamespace := os.Getenv("POD_NAMESPACE")
+
+	// When running inside the cluster namespace as cluster-installer, we only have access to Get & Update for ClusterDeployment
+	if podNamespace != "" {
+		clusterDeployment, err := hiveset.HiveV1().ClusterDeployments(podNamespace).Get(context.TODO(), podNamespace, v1.GetOptions{})
+		checkError(err)
+
+		takeAction(hiveset, kubeset, *clusterDeployment, TakeAction, true)
+		fmt.Println("  \\-> Event supressed")
+
+	} else {
+		// Grab all ClusterDeployments to change the state
+		cds, err := hiveset.HiveV1().ClusterDeployments(podNamespace).List(context.TODO(), v1.ListOptions{})
+		checkError(err)
+
+		for _, clusterDeployment := range cds.Items {
+
+			if (OptIn == "true" && clusterDeployment.Labels["hibernate"] == "true") || (OptIn != "true" && clusterDeployment.Labels["hibernate"] != "skip") {
+				takeAction(hiveset, kubeset, clusterDeployment, TakeAction, false)
 			} else {
 				fmt.Println("Skip    : " + clusterDeployment.Name + "  (currently " + string(clusterDeployment.Spec.PowerState) + ")")
-				fireEvent(kubeset, &clusterDeployment, "hibernating", "Skipping cluster "+clusterDeployment.Name+", requested powerState "+TakeAction+" equals current powerState "+string(clusterDeployment.Spec.PowerState), "skipHibernating", "Normal")
+				fireEvent(kubeset, &clusterDeployment, "hibernating", "Skipping cluster "+clusterDeployment.Name+", found labels.hibernate=skip. It will not be hibernating", "skipHibernating", "Normal")
+			}
+		}
+	}
+}
+
+func takeAction(hiveset *hiveclient.Clientset, kubeset *kubernetes.Clientset, clusterDeployment hivev1.ClusterDeployment, takeAction string, noEvents bool) {
+	if string(clusterDeployment.Spec.PowerState) != takeAction {
+
+		fmt.Print(takeAction + ": " + clusterDeployment.Name)
+		//newPowerState := powerStatePatch(hiveset, &clusterDeployment, TakeAction)
```

Comment:
Remove this?

Reply:
I want to leave it for now, as I would have preferred to use it. I added an additional comment to provide context and maybe someone will be able to use patch if the cluster-installer rbac changes down the road.